src/SIL.XForge.Identity/Controllers/Account/AccountController.cs

```diff
@@ -167,7 +179,92 @@ public async Task<IActionResult> Login(LoginInputModel model, string button)
             return View(vm);
         }
 
-        
+        /// <summary>
+        /// Show reset password page
+        /// </summary>
+        [HttpGet]
+        public async Task<IActionResult> ResetPassword(string token)
+        {
+            if (!string.IsNullOrEmpty(token))
+                return View();
+            else
+            {
+                LoginViewModel vm = await BuildLoginViewModelAsync("");
+                return View("Login", vm);
+            }
+        }
+
+        /// <summary>
+        /// Show forgot password page
+        /// </summary>
+        [HttpGet]
+        public async Task<IActionResult> ForgotPassword()
+        {
+            ForgotPasswordViewModel vm = BuildForgotPasswordViewModelAsync();
+            return View("ForgotPassword", vm);
+        }
+
+        /// <summary>
+        /// Validate username or email and send email
+        /// </summary>
+        [HttpPost]
+        public async Task<IActionResult> ForgotPassword(ForgotPasswordViewModel model)
+        {
+            UserEntity user = await _users.Query().SingleOrDefaultAsync(u => u.Username == model.UsernameOrEmail || u.Email == model.UsernameOrEmail);
+            if (user != null)
+            {
+                string emailId = "jamesprabud@gmail.com";
+                string subject = "Scripture Forge Forgotten Password Verification";
+
+                byte[] time = BitConverter.GetBytes(DateTime.UtcNow.ToBinary());
+                byte[] key = Guid.NewGuid().ToByteArray();
+                string ResetPasswordKey = Convert.ToBase64String(time.Concat(key).ToArray());
+
+                string body = "<div class=''><h1>Reset Password for " + user.Username + "</h1> " +
+                    "<p>Please click this link to <a href='https://beta.scriptureforge.local/account/resetpassword?token=" + ResetPasswordKey + "' target='_blank'>Reset Your Password</a>.</p> " +
+                    "<p>This link will be valid for 1 week only.</p><p>Regards,<br>The Scripture Forge team</p></div>";
+                SendEmail(subject, body, emailId);
+
+                user.ResetPasswordKey = ResetPasswordKey;
+                user.ResetPasswordExpirationDate = DateTime.Now.AddDays(7);
+                await _users.UpdateAsync(user);
+                LoginViewModel vm = await BuildLoginViewModelAsync("");
+                vm.ResetPasswordMessage = "Password Reset email sent for username " + model.UsernameOrEmail;
+                return View("Login", vm);
+            }
+            else
+            {
+                model.EnableErrorMessage = true;
+                return View("ForgotPassword", model);
+            }
+        }
+
+        public void SendEmail(string subject, string body, string emailId)
+        {
+            var siteOptions = _config.GetOptions<SiteOptions>();
```

Comment:
You should be able to use the `_options` field instead of getting a `SiteOptions` from `_config`. You should need `IConfiguration` at all, so that can be removed entirely.

Reply:
Yes, We removed config related code.